src/improvements/single.just

```diff
@@ -1,39 +1,40 @@
 set dotenv-load
 
-export rpcUrl := env_var('ETH_RPC_URL')
+# Common variables used across multiple recipes
+# Export variables in uppercase for shell script usage
+export TASK_PATH := invocation_directory()
+export SCRIPT_PATH := justfile_directory()
+
+# Get RPC URL based on task path
+rpc_cmd := SCRIPT_PATH + "/script/get-rpc-url.sh \"" + TASK_PATH + "\""
+export rpcUrl := shell(rpc_cmd)
+
+# Fetch the template name from the config.toml file
+cmd := "sed -n 's/^[[:space:]]*templateName[[:space:]]*=[[:space:]]*\\\"\\([^\\\"]*\\)\\\".*/\\1/p' " + TASK_PATH + "/config.toml"
+export SCRIPT_NAME := shell(cmd)
 export signatures := env_var_or_default('SIGNATURES', '')
-export taskPath := invocation_directory()
-export scriptPath := justfile_directory()
-export ownerSafe := env_var('OWNER_SAFE')
-export scriptName := env_var('SCRIPT_NAME')
 
 simulate hdPath='0':
   #!/usr/bin/env bash
-  config=${taskPath}/config.toml
-  script=${scriptPath}/template/${scriptName}.sol
+
+  config=${TASK_PATH}/config.toml
+  script=${SCRIPT_PATH}/template/${SCRIPT_NAME}.sol
 
   echo "Using script ${script}"
-  echo "Safe address: ${ownerSafe}"
-  echo "Getting signer address..."
-
-  if [ ! -z "$SIMULATE_WITHOUT_LEDGER" ]; then
-    signer=$(cast call ${ownerSafe} "getOwners()(address[])" -r ${rpcUrl} | grep -oE '0x[a-fA-F0-9]{40}' | head -n1)
-  else
-    signer=$(cast wallet address --ledger --mnemonic-derivation-path "m/44'/60'/{{hdPath}}'/0/0")
-  fi
-  echo "Simulating with: ${signer}"
   echo ""
 
   forge build
   forge script ${script} \
     --rpc-url ${rpcUrl} \
     --sig "simulateRun(string)" ${config} \
-    --sender ${signer}
```

Comment:
Same question, why was this deleted?

Reply:
Hm, this is in `single.just` so there are no child multisigs?

Reply:
So in the `single` flow we still use `approveHash`? If so I believe that is different from our current flow (cc @blmalone to confirm) so if we are changing that let's be really clear about the diff, it's consequences, and the rationale for it somewhere

Reply:
This seems to be resolved.

Reply:
I'm going to close this thread. Please DM if it shouldn't be closed. Lets follow up in a separate PR.